config/config.go

```diff
@@ -1327,15 +1324,15 @@ func (c *Config) buildAggregator(name string, tbl *ast.Table) (*models.Aggregato
 		Grace:  time.Second * 0,
 	}
 
-	c.getFieldDuration(tbl, "period", &conf.Period)
-	c.getFieldDuration(tbl, "delay", &conf.Delay)
-	c.getFieldDuration(tbl, "grace", &conf.Grace)
-	c.getFieldBool(tbl, "drop_original", &conf.DropOriginal)
-	c.getFieldString(tbl, "name_prefix", &conf.MeasurementPrefix)
-	c.getFieldString(tbl, "name_suffix", &conf.MeasurementSuffix)
-	c.getFieldString(tbl, "name_override", &conf.NameOverride)
-	c.getFieldString(tbl, "alias", &conf.Alias)
-	c.getFieldString(tbl, "log_level", &conf.LogLevel)
+	conf.Period = c.getFieldDuration(tbl, "period")
+	conf.Delay = c.getFieldDuration(tbl, "delay")
+	conf.Grace = c.getFieldDuration(tbl, "grace")
+	conf.DropOriginal = c.getFieldBool(tbl, "drop_original")
+	conf.MeasurementPrefix = c.getFieldString(tbl, "name_prefix")
+	conf.MeasurementSuffix = c.getFieldString(tbl, "name_suffix")
+	conf.NameOverride = c.getFieldString(tbl, "name_override")
+	conf.Alias = c.getFieldString(tbl, "alias")
+	conf.LogLevel = c.getFieldString(tbl, "log_level")
```

Comment:
I'm not convinced. The current version nicely shows the initialization of defaults first, followed by getting values parsed previously from the config.

Reply:
I have seen your update, then the same should apply also for the other types of `getField` methods..

Reply:
> then the same should apply also for the other types of `getField` methods..

It doesn't necessarily have to. I only changed it in the place where finding or not finding matters (because then we shouldn't override the previous value). For the rest of the cases, the methods return the zero value for the respective type (which is the same as before calling these methods).

Reply:
It doesn't matter for now indeed, however there might be introduced a new config option of one of the other types that has a default so needs this check as well. In that case it is handy that all similar functions already have the same behaviour and return values..

Reply:
We are talking about code that is not part of the plugin code but rather the global config code.
I'm confident that the people who might someday in the indefinite future need to adapt the code to a new field in the config will handle this "problem" just fine.